### PR TITLE
Update to .NET 6

### PR DESCRIPTION
**Pull request by GabrielChouinardLetourneau**
GitHub Issue: #

## Checklist

Please check if your PR fulfills the following requirements:

- [ ] Documentation has been added/updated
- [ ] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

File: MessageDialog.sln [hunk 1]
```diff
@@ -77,11 +84,36 @@ Global
 		{58D94E26-9A4A-45AD-9F5E-2FCB9B2F0EAE}.Release|NuGet.Build.0 = Release|Any CPU
 		{58D94E26-9A4A-45AD-9F5E-2FCB9B2F0EAE}.Release|UAP.ActiveCfg = Release|Any CPU
 		{58D94E26-9A4A-45AD-9F5E-2FCB9B2F0EAE}.Release|UAP.Build.0 = Release|Any CPU
+		{D20A05F5-1775-4EFB-9EBA-882B162F2BAA}.Debug|Android.ActiveCfg = Debug|Any CPU
```
Comment (Iguins) [inline MessageDialog.sln:87]: @carlh98  @GabrielChouinardLetourneau  needed?

**Review by Iguins — APPROVED** [review]

File: MessageDialog.sln (hunk 1 above)
Comment (GabrielChouinardLetourneau) [reply MessageDialog.sln:87]: Don't know if it is really necessary, I will resolve this conversation and let CH give his input

File: build/stage-build.yml [hunk 2]
```diff
@@ -19,16 +16,33 @@
     versionSpec: $(NUGET_VERSION)
     checkLatest: false
 
-- task: NuGetCommand@2
+- task: NuGetAuthenticate@1
```
Comment (carlh98) [inline build/stage-build.yml:19]: We can remove NuGetAuthenticate@1

File: src/MessageDialog.Uno.WinUI/MessageDialog.Uno.WinUI.csproj [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		<LangVersion>10.0</LangVersion>
+		<!-- Ensures the .xr.xml files are generated in a proper layout folder -->
+		<GenerateLibraryLayout>true</GenerateLibraryLayout>
+	</PropertyGroup>
+	<ItemGroup>
+		<PackageReference Include="MSBuild.Sdk.Extras" Version="3.0.44">
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+		</PackageReference>
+		<PackageReference Include="Uno.Core" Version="4.0.1" />
```
Comment (carlh98) [inline src/MessageDialog.Uno.WinUI/MessageDialog.Uno.WinUI.csproj:15]: Core is needed?

File: src/MessageDialog.Uno.WinUI/MessageDialog.Uno.WinUI.csproj [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		<PackageReference Include="Uno.WinUI" Version="4.4.13" />
+	</ItemGroup>
+	<ItemGroup Condition="'$(TargetFramework)'=='net6.0-windows10.0.18362'">
```
Comment (carlh98) [inline src/MessageDialog.Uno.WinUI/MessageDialog.Uno.WinUI.csproj:18]: I think RuntimeFrameworkVersion and TargetingPackVersion 22000 should be the same version framework as 18362

File: src/MessageDialog.Uno/MessageDialog.Uno.csproj [hunk 5]
```diff
@@ -19,11 +20,17 @@
 	</PropertyGroup>
 
 	<ItemGroup>
+		<PackageReference Include="MSBuild.Sdk.Extras" Version="3.0.44">
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+		</PackageReference>
 		<PackageReference Include="Uno.UI" Version="4.0.7" />
```
Comment (carlh98) [inline src/MessageDialog.Uno/MessageDialog.Uno.csproj:27]: same

**Pull request merged**